allow to suppress logging of completed requests (#36)
meatballhat/negroni-logrus: README.md

If you want to reuse an already initialized `logrus.Logger`, you should be using
`NewMiddlewareFromLogger` (see [existinglogrus](./examples/existinglogrus/example.go)).

## Logging

By default, starting requests and completed requests will be logged.
You are free to disable these log messages using the options `logStarting` and `logCompleted` of the `Middleware` struct.
The intention is to allow to replace these generic messages with custom messages more fitted to the usecase.

> Note that if logging completed requests is disabled and negroni-logrus is part of an entire chain of middlewares, logging the exact moment of the completed request is not possible.

Reviewer: I'm starting to think this is over-engineered. We'll end-up in messy code in the final apps with a risk of misplacing the last log call.
Since we already have an `middleware.After` [function](https://github.com/meatballhat/negroni-logrus/blob/master/middleware.go#L141), it should be used to achieve what you want.
It would be a breaking change, but what about move `[...].Info("completed handling request")
` to [`DefaultAfter`](https://github.com/meatballhat/negroni-logrus/blob/master/middleware.go#L162), so users can provide a complete function to achieve exactly what they want. It would be a cleaner code I think.
What do you think?
(sorry for the change of direction, I haven't touched this project since a while, and I forgot about this func).

Author: The change is breaking for people who provide a custom `middleware.After` function and rely on the existing `[...].Info("completed handling request")`. For those people, completed-messages will be gone.

My feeling is that people who replace `After` also log custom messages, i.e., the change is 'not so breaking'. Also, fixing the change is easy by simply adding an `Info(...)` to the custom `After`.

Imo, that change is ok.

Reviewer: I'm fine with a breaking change like this, it's just a matter of versioning after all. As you said, it will just remove the "completed handling request" text. My question was more, are you ok changing the behavior of this PR accordingly? Thanks

Author: Will do, however not until next week. Will report back.

If you want to reuse an already initialized `logrus.Logger`, you should be using
`NewMiddlewareFromLogger` (see [existinglogrus](./examples/existinglogrus/example.go)).

## Logging

> By default, starting requests and completed requests will be logged.

Reviewer: I'd like to change this, as it seems we're talking about two requests here. 

```suggestion
By default, requests will generate a log entry when starting and completing.
```
